pyrometheus/chem_expr.py

```diff
@@ -257,7 +278,15 @@ def falloff_function_expr(react: ct.Reaction, i, t, red_pressure, falloff_center
         of the temperature *t*, reduced pressure *red_pressure*, and falloff center
         *falloff_center* as a :class:`pymbolic.primitives.Expression`
     """
-    if react.falloff.falloff_type == "Troe":
+    if react.uses_legacy:
+        falloff_type = react.falloff.falloff_type
+        from warnings import warn
+        warn("New Cantera implementation retrieves 'FalloffRate' objects "
+             "using the rate 'property'", DeprecationWarning)
+    else:
+        falloff_type = react.rate.type
+
+    if falloff_type == "Troe":
```

Comment:
I know that this is old code, but this is not a good pattern. Instead, act only on expected values and raise an error on unexpected ones.

Reply:
Sorry to ask for further clarification, but does this mean that I should not act (generate expressions) on legacy (cti) reaction patterns? That would lead to mirgecom test failure, unless mirgecom itself has already been updated too. I think there is something I'm still missing.

Reply:
What I mean here is the following pattern:
```py
if falloff_type == "some_known_value":
    do the thing
else:
    # silently assume it's something else
    do the other thing
```
A more defensive way to do the same thing:
```py
if falloff_type == "some_known_value":
    do the thing
elif falloff_type == "another_known_value":
    do the other thing
else:
    raise ValueError(f"unexpected value of 'falloff_type': '{falloff_type}'")
```